src/components/pill/pill.component.js

```diff
@@ -51,6 +57,8 @@ class Pill extends React.Component {
 }
 
 Pill.propTypes = {
+  /** Styled system margin spacing props */
+  ...Pill.getMarginSpacePropTypes(),
```

Comment:
It's not required here because we're adding spacing props tables through the `StyledSystemProps` mdx block from `.storybook/utils/styled-system-props`

Check https://github.com/Sage/carbon/pull/3616/files#diff-c766022507dfdd4a9b51cc9392a792e9e9e3a70c51c6d9efa5bfcaa57332a236

Reply:
But we still need it for props validation don't we?